flink-core/src/main/java/org/apache/flink/api/common/functions/RuntimeContext.java

```diff
@@ -101,9 +101,9 @@
 
 	/**
 	 * For system internal usage only. Use getAccumulator(...) to obtain a
-	 * accumulator. Use this as read-only.
+	 * accumulator. The returned map must NOT be modified.
 	 */
-	HashMap<String, Accumulator<?, ?>> getAllAccumulators();
+	Map<String, Accumulator<?, ?>> getAllAccumulators();
```

Comment:
No, this is a user-facing to get the Accumulator map in the RichFunction. I changed the return type to Map and I'm returning a Collections.immutableMap to prevent modifications. I think it makes sense that the user can get a list of registered accumulators and their current status (e.g. in chained operators).